src/NServiceBus.RabbitMQ/Configuration/ConnectionConfiguration.cs

```diff
@@ -2,58 +2,145 @@
 {
     using System;
     using System.Collections.Generic;
+    using System.Collections.ObjectModel;
+    using System.Data.Common;
     using System.Diagnostics;
     using System.IO;
+    using System.Linq;
+    using System.Text;
+    using Logging;
     using Support;
 
     class ConnectionConfiguration
     {
-        public string Host { get; set; }
+        static readonly ILog Logger = LogManager.GetLogger(typeof(ConnectionConfiguration));
 
-        public int Port { get; set; }
+        const bool defaultUseTls = false;
+        const int defaultPort = 5672;
+        const int defaultTlsPort = 5671;
+        const ushort defaultRequestedHeartbeat = 5;
+        static readonly TimeSpan defaultRetryDelay = TimeSpan.FromSeconds(10);
+        const string defaultCertPath = "";
+        const string defaultCertPassphrase = null;
 
-        public string VirtualHost { get; set; }
+        public string Host { get; }
 
-        public string UserName { get; set; }
+        public int Port { get; }
 
-        public string Password { get; set; }
+        public string VirtualHost { get; }
 
-        public ushort RequestedHeartbeat { get; set; }
+        public string UserName { get; }
 
-        public TimeSpan RetryDelay { get; set; }
+        public string Password { get; }
 
-        public bool UseTls { get; set; }
+        public ushort RequestedHeartbeat { get; }
 
-        public string CertPath { get; set; }
+        public TimeSpan RetryDelay { get; }
 
-        public string CertPassphrase { get; set; }
+        public bool UseTls { get; }
 
-        public Dictionary<string, object> ClientProperties { get; } = new Dictionary<string, object>();
+        public string CertPath { get; }
 
-        public ConnectionConfiguration(string endpointName)
+        public string CertPassphrase { get; }
+
+        public IReadOnlyDictionary<string, string> ClientProperties { get; }
```

Comment:
Yeah, come to think of it, I agree. I'll send a fixup!

The commit that added this has the message "made ConnectionConfiguration immutable" but when I squash I'll alter the message, since it won't be immutable anymore. That said, I'm not sure if we want to keep the micro commits or not before merging.